dask/array/overlap.py

```diff
@@ -413,15 +416,24 @@ def overlap(x, depth, boundary):
            [100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100],
            [100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100]])
     """
+    # import pdb; pdb.set_trace()
     depth2 = coerce_depth(x.ndim, depth)
     boundary2 = coerce_boundary(x.ndim, boundary)
 
-    # rechunk if new chunks are needed to fit depth in every chunk
     depths = [max(d) if isinstance(d, tuple) else d for d in depth2.values()]
     new_chunks = tuple(
         ensure_minimum_chunksize(size, c) for size, c in zip(depths, x.chunks)
     )
-    x1 = x.rechunk(new_chunks)  # this is a no-op if x.chunks == new_chunks
+    if allow_rechunk is True:
+        # rechunk if new chunks are needed to fit depth in every chunk
+        x1 = x.rechunk(new_chunks)  # this is a no-op if x.chunks == new_chunks
+    else:
+        raise ValueError(
```

Comment:
This raises any time allow_rechunk is False.  It should only raise if the smallest chunk is smaller than the depth (in any dimension).